### ISSUE TITLE
Using eslint-plugin-jsx-a11y

### ISSUE THREAD

**Issue by fkereki** [evcohen/eslint-plugin-jsx-a11y#466]
If you use Styled-Components, eslint-plugin-jsx-a11y cannot detect tags and apply rules, so it's rendered practically useless.

const Styled_Div = styled.div`...`;
const Nice_Button = styled.button`...`;

render() {
    return <Styled_Div... />...<Nice_Button...> ... </Nice_Button>...</Styled_Div>;
}
I have two suggestions:

add some definitions for equivalences in the configuration file, at the same level of plugins and rules, like:
equivalences = {
    'Styled_Div' : 'div',
    'Nice_Button' : 'button'
}
if the name of an tag ends with underscore and a HTML tag (as in my examples) just assume the tag: Styled_Div would be processed as a div, and Nice_Button as a button.

**Comment by ljharb** [comment]
I think that "rendered practically useless" is true of just about any static analysis tool for JS, when using a template tag DSL like styled-components.

Regardless, I can see the benefit (even for idiomatic jsx users) of specifying a settings-level map - from `'div'` to `['Styled_Div', 'OtherDiv']` etc - that all rules used.

This wouldn't be a trivial amount of work - and would require lots of tests, for every single rule - but I think it'd be useful.

**Comment by fkereki** [comment]
Wouldn't my second suggestion be simpler to implement?

Best regards,
FK

**Comment by ljharb** [comment]
It would be much more magical and implicit, which are synonyms for "worse" imo.

**Comment by fkereki** [comment]
I can understand that - however, as an interim measure, it could do: "better a bird in the hand, than two in the bush"? Thanks for your answers!

**Comment by garetmckinley** [comment]
Any update on this? Most of my projects use styled-components pretty heavily and it's a serious bummer missing out on getting a11y hints

**Comment by eps1lon** [comment]
Looks like #174 could've solved this issue.

**Comment by brendanmorrell** [comment]
Hey, I made a pretty cool plugin for this (hopefully). It found 6,567 errors in one of my repos that were undiscovered using airbnb/jsx-a11y rules. Please give it a star so we can get more people interested, using it, and contributing to make a more accessible internet for all.

Here’s the [github repo](https://github.com/brendanmorrell/eslint-plugin-styled-components-a11y) it covers all 4 styled components/emotion syntaxes

Existing a11y linting has become significantly less useful (personally, 99% of the components I need linted are styled.). Each scenario below would display the error
```
Visible, non-interactive elements with click handlers must have at least one keyboard listener.`
```

```
const Div = styled.div``;
<Div onClick={() => null}/>
```

```
const Div = styled.div.attrs({ onClick: () => null})``;
<Div />
```

```
const Div = styled.div.attrs({ onClick: () => null})``;
const StyledDiv = styled(Div)``;
<StyledDiv />
```

```
const ButtonAsDiv = styled.button``;
<ButtonAsDiv as="div" onClick={() => null} />
```


It’s not perfect (no linters are), but it’s super useful, and with a little time and community support, it could be even better.
